sycl/plugins/esimd_cpu/esimd_libcm.hpp

```diff
@@ -0,0 +1,359 @@
+#include <CL/sycl/detail/kernel_desc.hpp>
+#include <CL/sycl/detail/type_traits.hpp>
+#include <CL/sycl/group.hpp>
+#include <CL/sycl/id.hpp>
+#include <CL/sycl/kernel.hpp>
+#include <CL/sycl/nd_item.hpp>
+#include <CL/sycl/range.hpp>
+
+#include <esimdcpu_runtime.h>
+
+#include <functional>
+#include <thread>
+#include <vector>
+
+#define LAMBDA_WRAPPER_TMPL(ARGTYPE, TAG)                                      \
+  typedef std::function<void(const ARGTYPE &)> LambdaFunction_##TAG;           \
+                                                                               \
+  extern "C" struct LambdaWrapper_##TAG {                                      \
+    LambdaFunction_##TAG f;                                                    \
+    std::vector<ARGTYPE> argVector;                                            \
+  };                                                                           \
+                                                                               \
+  template <typename LambdaTy>                                                 \
+  LambdaWrapper_##TAG *makeWrapper_##TAG(std::vector<ARGTYPE> &vec,            \
+                                         LambdaTy f) {                         \
+    LambdaWrapper_##TAG *w = new LambdaWrapper_##TAG;                          \
+    w->f = LambdaFunction_##TAG(f);                                            \
+    w->argVector = vec;                                                        \
+    return w;                                                                  \
+  }                                                                            \
+                                                                               \
+  extern "C" inline void invokeLambda_##TAG(void *p) {                         \
+    auto *w = reinterpret_cast<LambdaWrapper_##TAG *>(p);                      \
+    w->f(w->argVector[cm_support::thread_local_idx()]);                        \
+  }
+
+#define _COMMA_ ,
+
+LAMBDA_WRAPPER_TMPL(sycl::id<3>, ID_3DIM)
+LAMBDA_WRAPPER_TMPL(sycl::id<2>, ID_2DIM)
+LAMBDA_WRAPPER_TMPL(sycl::id<1>, ID_1DIM)
+LAMBDA_WRAPPER_TMPL(sycl::item<3 _COMMA_ false>, ITEM_3DIM)
+LAMBDA_WRAPPER_TMPL(sycl::item<2 _COMMA_ false>, ITEM_2DIM)
+LAMBDA_WRAPPER_TMPL(sycl::item<1 _COMMA_ false>, ITEM_1DIM)
+LAMBDA_WRAPPER_TMPL(sycl::item<3 _COMMA_ true>, ITEM_OFFSET_3DIM)
+LAMBDA_WRAPPER_TMPL(sycl::item<2 _COMMA_ true>, ITEM_OFFSET_2DIM)
+LAMBDA_WRAPPER_TMPL(sycl::item<1 _COMMA_ true>, ITEM_OFFSET_1DIM)
+LAMBDA_WRAPPER_TMPL(sycl::nd_item<3>, NDITEM_3DIM)
+LAMBDA_WRAPPER_TMPL(sycl::nd_item<2>, NDITEM_2DIM)
+LAMBDA_WRAPPER_TMPL(sycl::nd_item<1>, NDITEM_1DIM)
+
+#undef LAMBDA_WRAPPER_TMPL
+#undef _COMMA_
+
+template <class KernelType, class KernelArgType, int __Dims__>
+class libCMBatch {
+private:
+  KernelType MKernel;
+  std::vector<KernelArgType> ArgVector;
+  std::vector<uint32_t> SpaceDim;
+  const std::vector<uint32_t> singleGrpDim = {1, 1, 1};
```

Comment:
```suggestion
  const std::vector<uint32_t> SingleGrpDim = {1, 1, 1};
```
Shouldn't it be SingleWIDim?